cli/cppcheckexecutor.cpp

```diff
@@ -129,6 +129,10 @@ namespace {
             return !mCriticalErrors.empty();
         }
 
+        const std::string& getCtuInfo() const {
+            return mCtuInfo;
```

Comment:
It feels a bit strange that this lives in the just remotely related logger. I wonder if all the logic should live in its own object and cleanup should be RAII-based and such.

That would be similar to what I tried in #6634 which comes up short in making it a local object because it needs to intercept the logged error at some point (which also applies to the code here). So adding some kind of hook into the error logging which std, XML and CTU (and even SARIF) could use would make a lot of sense.

Reply:
I am not sure if I understand these comments fully.

I do feel the ctu info belongs in cli and gui rather than lib. it's aggregated info from all threads. For me it's reasonable that CppcheckExecutor owns the data (directly or indirectly) however I can agree that the functionality does not technically belong in StdLogger actually.

We also have the mActiveCheckers and mCriticalErrors. Those do not really fit very well in the StdLogger neither.

I do not see a very elegant way to insert hooks. We could add one more ErrorLogger that owns the StdLogger?

     class AllLogger : public ErrorLogger {
     private:
         void reportErr(const ErrorMessage& errmsg) {
              ... handle active checkers, critical errors, ctu, ...

              // pass remaining errors to stdLogger
              stdLogger.reportErr(errmsg);
         }
         StdLogger stdLogger;
    };

Do you have better ideas..?

Reply:
Personally I still suggest to put the condition in StdLogger that ensures the output is formatted with sarif/text/xml so it will take a ErrorMessage input and output the proper string..

Reply:
Yeah, again something in this file turns into a grab-bag after things got sorted out at some point.

This is not something we need to solve in this PR. I will see if I am put together a PR with the proposed hook based on the previously linked cleanup.